### DISCUSSION TITLE
xDrip sync to InfluxDB

### DISCUSSION THREAD

**Discussion by BorisBertelsen** [NightscoutFoundation/xDrip#2929] [General]
Cheers everyone,
I'm a completely new to sensors and xDrip so bear with me.

I'm trying to sync data to my local influxDB so I can show my most recent bloodsugar on a little screen by my bed, that way I don't have to check my phone or watch to see what the levels are.

I have some problems syncing the data to my local influxDB. I wonder if I'm missing a specific setting or doing something completely wrong. Here's a quicky on what I have done so far:

- My influxDB is running locally at http://192.168.0.31:8086
- I have done nothing else in InfluxDB beside creating a Bucket
- In xDrip > Cloud upload > InfluxDB I have enabled sync. My InfluxDB URI is the http://192.168.0.31:8086, Database name, user and password is also entered
- I have "Skip LAN uploads" so I only upload onces I'm home on my own network
- I also installed Grafana locally to see some graphs but also **verify that data is available in influxDB** but I don't get anything when I connect Grafana to influxDB. There's no data.

My question is: How do I verify that xDrip is actually sending data to my local influxDB?

Bests,
Mike

**Comment by martinachtman** [comment]
Check [mlukasek/M5_NightscoutMon](https://github.com/mlukasek/M5_NightscoutMon/releases) which connects directly to your nightscout instance rather than influxDB.

**Comment by Navid200** [comment]
I cannot answer your questions.
But, these may be relevant.

You can set the screensaver of your phone to be xDrip.  Then, when the phone is being charged, the screen will show the xDrip graph continuously.
https://navid200.github.io/xDrip/docs/Screensaver.html

If your phone has the "Always on display" feature, you can set it to show the xDrip widget when the phone screen locks (not charging).
https://navid200.github.io/xDrip/docs/Display/AOD.html

There may be stands that could hold your phone on the bedside table with an angle so that you could see the screen when lying in bed.

**Comment by jamorham** [comment]
I think the influxdb client library that is in xDrip is quite old and probably doesn't work with newer installations. It needs updating I expect.

**Comment by BorisBertelsen** [comment]
Hey guys and thank you for reading my ticket.
I learned something new. xDrip has this "View Events Log" where it tell me it cannot write to influxDB because it's unauthorized.

Have I missed something in my setup perhaps? InfluxDB settings in xDrip+ is all fulfilled and I can't find anywhere in InfluxDB interface where I can change security settings.

Any leads?

**Reply by Navid200** [reply]
I hope you saw JamOrHam's response.  Based on that, this needs some work.  So, I don't know why you are asking again.  We are volunteers.
If you want, you can convert this to an issue by clicking on "Create issue from discussion" at the bottom right.  
But, you have to be patient.
And there is no guarantee it will be fixed, or, it will be fixed right away.

**Reply by BorisBertelsen** [reply]
Hi Navid,

I saw all the comments. The reason why I'm asking again is because of two things:
1) He "thinks" the library is outdated
2) I found the event log which prompted me a message about xDrip not being authorized to write to InfluxDB

Because of the event log message I thought it might not be related to an outdated dependency but rather a matter of security options somewhere in xDrip or Influx.

By asking again about the event log message I could maybe eliminate the possibility of the dependency being outdated.

I do find myself patient. I'm not rushing anything and I'm not bumping my message. I'm trying to figure out why xDrip is unauthorized so myself and anyone else finding this post can learn something.

I do appreciate all the volunteers and I have much love for the fact people are willing to use their freetime working on this project. I'm here to learn and hopefully put my own skills to good use. But I have to ask, to learn.

I'll see if I can figure it out and update the post accordingly.

**Reply by jamorham** [reply]
To resolve the problem we would need something to test it against. Is there a public influx database that can be used or can you provide quickstart instructions for installing something similar to the setup you are using? 

Developer steps needed would be to reproduce the problem, investigate the cause, find a resolution, locally test it and then deploy a nightly for wider testing to confirm. As none of the core developers are using influxdb and are not familiar with it we're unable to reproduce the problem. Please provide some more guidance to help with step 1 and the remaining steps may be quite easy.

**Comment by jamorham** [comment]
@Sp34k Hi, I've tested this and it appears to work fine. Is the problem that you are using influxdb2 and not influxdb ?
